client/src/pages/contentPages/Groups.js

```diff
@@ -1,41 +1,326 @@
+
+  const classes = useStyles();
+  const { userCourse, profile } = useGlobalContext();
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [modalStyle] = React.useState(getModalStyle);
+  const [openModal, setOpen] = useState(false);
+  const [courses, setCourses] = useState([]);
+  const [groupName, setGroupName] = useState('');
+  const [courseId, setCourseId] = useState('');
+  const [userProfile, setProfile] = useState([])
+  const [groupError, setGroupErrors] = useState('');
+  const [groupNames, setGroupNames] = useState([]);
+
+  const [uploading, setUploading] = useState(false);
+  const [groupPicture, setGroupPicture] = useState('');
+  const [formValid, setFormValid] = useState(false)
+
+  const onDrop = useCallback(async (droppedFiles) => {
+    if (droppedFiles.length) {
+        setUploading(true);
+
+        const form = new FormData();
+        form.append('image', droppedFiles[0]);
+        const res = await axios.post('/upload/single', form)
+          .catch((err) => console.log(err));
+        if (res && res.data) {
+          setGroupPicture(res.data.imageUrl)
+          setUploading(false)
+        }
+    }
+  }, []);
+
+  const {
+    getRootProps,
+    isDragActive,
+    isDragAccept,
+    isDragReject,
+  } = useDropzone({onDrop, maxFiles: 1, accept: 'image/*'});
+
+  const style = useMemo(() => ({
+    ...baseStyle,
+    ...(isDragActive ? activeStyle : {}),
+    ...(isDragAccept ? acceptStyle : {}),
+    ...(isDragReject ? rejectStyle : {})
+  }), [
+    isDragActive,
+    isDragReject,
+    isDragAccept
+  ]);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setGroupPicture('');
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setGroupPicture('');
+    setGroupName('');
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    
+    if(formValid) {
+
+      const data = { groupName: groupName, imageUrl: groupPicture , courseId: courseId};
+      const response = await axios.post('/user/groups', data);
+
+      if(response.data) {
+        enqueueSnackbar('Group Created Successfully.', {
+          variant: 'success',
+          autoHideDuration: '5000',
+        });
+        setOpen(false)
+      } else {
+        enqueueSnackbar('Error Creating Group.', {
+          variant: 'Error',
+          autoHideDuration: '5000',
+        });
+      }
+      
+    } else {
+      enqueueSnackbar('Error Creating Group.', {
+          variant: 'Error',
+          autoHideDuration: '5000',
+      });
+    }
+
+  };
+
+  const handleOnChange = async (e) => {
+    checkGroupExists(e.target.value.toLowerCase())
+    
+  }
+
+const checkGroupExists = async (name) => {
+  try {
+
+    setGroupName(name)
+
+    if(groupNames.includes(name) !== true) {
+      setGroupErrors('');
+      setFormValid(true);
+      return false;
+    } else {
+      setGroupErrors("A group with this name already exists!");
+      setFormValid(false);
+      return true;
+    }
+
```

Comment:
It looks like this function is checking the users groups for uniqueness. I think it needs to compare against all the groups in the course that's selected. But I may be misremembering what we decided on. If that is the case, you can get all the groups of all the courses the user is in by accessing  userGroups.courseGroups in context.